Modify useTitle.ts' logic

zsh2401 opened the pull request:
Restore the previous title when the current component is destroyed.

FlickerLogicalStack commented on src/useTitle.ts:
```diff
@@ -1,11 +1,14 @@
 import { useRef } from 'react';
 
 function useTitle(title: string) {
```
I think upgraded `useTitle` can be like:

```suggestion
import { useEffect, useRef } from 'react';

type Options = {
  restoreOnUnmount?: boolean;
};

function useTitle(
  title: string,
  { restoreOnUnmount = false }: Options = {}
) {
  const prevTitle = useRef(document.title);

  document.title = title;

  useEffect(() => {
    if (restoreOnUnmount) {
      return () => {
        document.title = prevTitle.current;
      };
    }
  }, []);
}

```

The main point in fact that restoration must be optional

streamich replied:
@zsh2401 can we make title restoration on un-mount optional, like @FlickerLogicalStack proposed, so this change is not breaking?

zsh2401 replied:
@streamich I DO AGREE! I will do it as soon as possible!

zsh2401 commented:
@streamich  ALL DONE! I have modified code so that it could restore previous title optionally.I have also renamed 'useTitle.test.tsx' to 'useTitle.test.ts', because it's looks like not right somehow.

streamich commented:
:tada: This PR is included in version 13.16.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/react-use)
- [GitHub release](https://github.com/streamich/react-use/releases/tag/v13.16.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
